### PR TITLE
More hardfork cleanup

### PR DESCRIPTION
**Pull request by jmjatlanta**
This is a continuation of #1743 

Finishes that ticket, and cleans hardforks from 3.0.1

### CODE REVIEW

File: libraries/chain/db_maint.cpp [hunk 1]
```diff
@@ -943,33 +943,6 @@ void database::process_bitassets()
...
-   {
-      const auto& current_asset = *asset_itr;
-      graphene::chain::share_type current_supply = current_asset.dynamic_data(db).current_supply;
-      graphene::chain::share_type max_supply = current_asset.options.max_supply;
-      if (current_supply > max_supply && max_supply != GRAPHENE_MAX_SHARE_SUPPLY)
-      {
-         wlog( "Adjusting max_supply of ${asset} because current_supply (${current_supply}) is greater than ${old}.", 
-               ("asset", current_asset.symbol) 
-               ("current_supply", current_supply.value)
-               ("old", max_supply));
```
Comment (abitmore) [inline libraries/chain/db_maint.cpp]: Do we have any asset with `current_supply > GRAPHENE_MAX_SHARE_SUPPLY`? If yes, we can leave a warning; if no, we can leave a comment somewhere.

File: libraries/chain/market_evaluator.cpp [hunk 2]
```diff
@@ -168,19 +168,8 @@ void_result call_order_update_evaluator::do_evaluate(const call_order_update_ope
...
-    * up the softfork code IF it never executes. We MAY be able to turn the hardfork code into regular code IF
-    * noone ever attempted this before HF 1465.
-    */
-   if (next_maintenance_time <= SOFTFORK_CORE_1465_TIME)
-   {
-      if ( _dynamic_data_obj->current_supply + o.delta_debt.amount > _debt_asset->options.max_supply )
-         ilog("Issue 1465... Borrowing and exceeding MAX_SUPPLY. Will be corrected at hardfork time.");
-   }
-   else
+   // We cannot remove this, as there were (at least 3) instances of this happening before the hardfork
```
Comment (abitmore) [inline libraries/chain/market_evaluator.cpp]: But they got corrected later, so we can remove the one-time processing code? Please leave a comment.

File: libraries/chain/market_evaluator.cpp [hunk 3]
```diff
@@ -325,6 +314,7 @@ object_id_type call_order_update_evaluator::do_apply(const call_order_update_ope
          call_obj = d.find(call_order_id);
          // we know no black swan event has occurred
          FC_ASSERT( call_obj, "no margin call was executed and yet the call object was deleted" );
+         // this HF must remain as-is, as the assert inside the "if" was triggered
```
Comment (abitmore) [inline libraries/chain/market_evaluator.cpp]: ... during `push_proposal()`.

File: tests/tests/market_fee_sharing_tests.cpp [hunk 4]
```diff
@@ -576,15 +535,6 @@ BOOST_AUTO_TEST_CASE(create_asset_via_proposal_test)
...
-      {
-         signed_transaction tx;
-         tx.operations.push_back( prop );
-         db.current_fee_schedule().set_fee( tx.operations.back() );
-         set_expiration( db, tx );
-         sign( tx, issuer_private_key );
-         GRAPHENE_CHECK_THROW(PUSH_TX( db, tx ), fc::exception);
-      }
-
       generate_blocks_past_hf1268();
```
Comment (abitmore) [inline tests/tests/market_fee_sharing_tests.cpp]: I thought we had removed all hf check code about hf1268, why leave it here then?

Comment (jmjatlanta) [reply tests/tests/market_fee_sharing_tests.cpp]: Unfortunately, some must remain. See db_market.cpp. pay_market_fees() needs an extra parameter after the hf.

File: libraries/chain/market_evaluator.cpp (hunk 2 above)
Comment (jmjatlanta) [reply libraries/chain/market_evaluator.cpp]: Replays with blocks before the hf will still need this I believe.

**Comment by pmconrad** [comment]
`chain_test` failing in travis

File: tests/tests/market_fee_sharing_tests.cpp (hunk 4 above)
Comment (abitmore) [reply tests/tests/market_fee_sharing_tests.cpp]: I don't think the parameter is necessary. Before the hard fork, the fee sharing field is always unset, so theoretically it will work without the parameter. A minor code refactory would be required.

Comment (jmjatlanta) [reply tests/tests/market_fee_sharing_tests.cpp]: pay_market_fees is called by

- fill_settle_order, which always calls the old version of the method
- fill_limit_order, which calls the correct version of the method, depending on head_block_time().

Both versions of the method check to assure that there are issuer_fees to be paid before going any further. The new version includes a section to determine if rewards are allowed. If they are not allowed, it does what the old version does. My suggestion for refactoring would be to add logic to is_rewards_allowed to check that seller != graphene::chain::account_object(), and modify both callers to always call the new method.

As for performance, this will avoid the head_block_time() call, the comparison with HARDFORK_1268_TIME, but add a comparison within pay_market_fees().

1. Does my suggestion for refactoring above make sense?
1. Is there a better way of doing the seller comparison ( i.e. avoid the temporary object creation) while maintaining readability?

File: libraries/chain/db_market.cpp [hunk 5]
```diff
@@ -931,7 +929,7 @@ bool database::fill_settle_order( const force_settlement_object& settle, const a
 { try {
    bool filled = false;
 
-   auto issuer_fees = pay_market_fees(get(receives.asset_id), receives);
+   auto issuer_fees = pay_market_fees( account_object(), get(receives.asset_id), receives);
```
[the remark this answers is not in the document]
Comment (jmjatlanta) [reply libraries/chain/db_market.cpp]: That is fine. Would you recommend a std::shared_ptr or an fc::optional?

[the remark this answers is not in the document]
Comment (pmconrad) [reply libraries/chain/db_market.cpp]: All of these require an object construction, albeit slightly cheaper than an `account_object`.
Alternative: define a local `static const account_object EMPTY;` and pass that.

[the remark this answers is not in the document]
Comment (abitmore) [reply libraries/chain/db_market.cpp]: `account_object()` has same `id` as `committee-account`, which is another reason that I'd like to avoid using it.

[the remark this answers is not in the document]
Comment (jmjatlanta) [reply libraries/chain/db_market.cpp]: While I believe avoiding raw pointers is a good idea, I implemented one here for the following reasons:
1. Limited scope. This method is currently only used in 1 source file.
1. No constructed objects (probably a minor performance improvement).
1. Avoids confusion with committee-account.
1. Easy comparison.

Feel free to shoot holes in my reasoning above, and as always, the code.

File: libraries/chain/db_market.cpp [hunk 6]
```diff
@@ -929,7 +929,7 @@ bool database::fill_settle_order( const force_settlement_object& settle, const a
 { try {
    bool filled = false;
 
-   auto issuer_fees = pay_market_fees( account_object(), get(receives.asset_id), receives);
+   auto issuer_fees = pay_market_fees( nullptr, get(receives.asset_id), receives);
```
Comment (abitmore) [inline libraries/chain/db_market.cpp]: I still didn't get why settle orders don't share market fee. IMHO it's a bug. Need a hardfork to fix it though. Will create a new issue.

**Comment by abitmore** [comment]
@jmjatlanta Can you rebase and remove these 2 commits from history?
* Remove parts of 1465
* Revert "Remove parts of 1465"

Best if squash the last commit into the 2nd last commit as well.

Thanks.

**Comment by pmconrad** [comment]
Your excel sheet is undecided about core-583.
Also, core-588 and core-604 are marked as 'Probably', please check and remove if possible.

**Comment by jmjatlanta** [comment]
I updated the spreadsheet and code for 583, 588, and 604.

File: libraries/chain/db_market.cpp [hunk 7]
```diff
@@ -641,7 +640,7 @@ int database::match( const limit_order_object& bid, const call_order_object& ask
    bool before_core_hardfork_184 = ( maint_time <= HARDFORK_CORE_184_TIME ); // something-for-nothing
    // TODO remove when we're sure it's always false
```
Comment (pmconrad) [inline libraries/chain/db_market.cpp]: What about this TODO?

**Comment by pmconrad** [comment]
Snapshot comparison @38M blocks is good.

File: libraries/chain/db_market.cpp (hunk 7 above)
Comment (jmjatlanta) [reply libraries/chain/db_market.cpp]: I was able to clean up the warnings for 342 and 184, as they never appeared in the log for both testnet nor mainnet.

**Comment by jmjatlanta** [comment]
@abitmore @pmconrad Please take another look at this PR and let me know if there are any outstanding issues. I think I've got them all covered. Thanks!

File: libraries/chain/asset_evaluator.cpp [hunk 8]
```diff
@@ -466,6 +411,7 @@ void_result asset_update_bitasset_evaluator::do_evaluate(const asset_update_bita
 
    FC_ASSERT( !current_bitasset_data.has_settlement(), "Cannot update a bitasset after a global settlement has executed" );
 
+   // hf 922_931 is a consensus/logic change. This hf cannot be removed.
```
Comment (abitmore) [inline libraries/chain/asset_evaluator.cpp]: Which logic change below is triggered by which asset(s) at which block(s)? Mainnet or testnet?

Comment (jmjatlanta) [reply libraries/chain/asset_evaluator.cpp]: I hope this answers your question. I added an "else" to this "if" and logged if the test failed. At 16% replayed on mainnet, I grep'd the following from the logs:

```2019-09-05T19:07:46        th_a:?unnamed?     cond_wlog_assert ] Before HF 922/931: Cannot update an asset to be backed by itself. at block 20231047			asset_evaluator.cpp:404
2019-09-05T19:18:25        th_a:?unnamed?     cond_wlog_assert ] Before HF 922/931: Cannot update an asset to be backed by itself. at block 23721657			asset_evaluator.cpp:404
2019-09-05T19:20:33        th_a:?unnamed?     cond_wlog_assert ] Before HF 922/931: Cannot update an asset to be backed by itself. at block 24080299			asset_evaluator.cpp:404
2019-09-05T19:21:48        th_a:?unnamed?     cond_wlog_assert ] Before HF 922/931: Cannot update an asset to be backed by itself. at block 24276618			asset_evaluator.cpp:404
2019-09-05T19:21:48        th_a:?unnamed?     cond_wlog_assert ] Before HF 922/931: A BitAsset cannot be backed by a BitAsset that itself is backed by a BitAsset. at block 24276618			asset_evaluator.cpp:404
2019-09-05T19:22:05        th_a:?unnamed?     cond_wlog_assert ] Before HF 922/931: Cannot update an asset to be backed by itself. at block 24328794			asset_evaluator.cpp:404
```

I believe this shows that by removing the hardfork code, consensus would not be maintained. Let me know if you need more information.

Comment (abitmore) [reply libraries/chain/asset_evaluator.cpp]: Thanks.

**Comment by abitmore** [comment]
Looks good so far, although there are still some warnings that can be removed.

**Comment by jmjatlanta** [comment]
I found that one warning relating to HF 935 did not appear in mainnet logs. `process_hf_935 changed something` ... I will check testnet and update here.

**Comment by jmjatlanta** [comment]
`process_hf_935 changed something` does not appear in mainnet or testnet. It looks as if the method `process_hf_935` can be removed. But there is a comment that makes me believe the testnet environment works a bit different, and this method should perhaps remain. 

https://github.com/bitshares/bitshares-core/blob/ca8e352686b36c21a442a1dbfeba67d639b90488/libraries/chain/db_maint.cpp#L1100L1113

**Comment by pmconrad** [comment]
If I understand the comments in #935 correctly we had a bug in testnet that was fixed before the code was moved into mainnet.
I think the comment indicates that there's a theoretical possibility that the bug could have been triggered in testnet. In that case the code couldn't be removed.

IMO the absence of the message proves that the code need not run and can be removed.

**Comment by jmjatlanta** [comment]
> IMO the absence of the message proves that the code need not run and can be removed.

I will remove the code and run snapshots. That should prove it. Thanks!

File: libraries/chain/proposal_evaluator.cpp [hunk 9]
```diff
@@ -206,7 +206,7 @@ object_id_type proposal_create_evaluator::do_apply(const proposal_create_operati
          top.to = GRAPHENE_RELAXED_COMMITTEE_ACCOUNT;
          top.amount = asset( GRAPHENE_MAX_SHARE_SUPPLY );
          proposal.proposed_transaction.operations.emplace_back( top );
-         wlog( "Issue 1479: ${p}", ("p",proposal) );
+         wlog( "Issue 1479: ${p}", ("p",proposal) ); // Note: This happened on mainnet 1.10.17503
```
Comment (abitmore) [inline libraries/chain/proposal_evaluator.cpp]: Perhaps directly replace the warning with that comment?

File: libraries/chain/db_maint.cpp [hunk 10]
```diff
@@ -1061,25 +1057,13 @@ void process_hf_868_890( database& db, bool skip_check_call_orders )
          }
       } // end loop of each feed
 
-      // if any feed was modified, print a warning message
-      if( feeds_changed )
-      {
-         wlog( "Found invalid feed for asset ${asset_sym} (${asset_id}) during hardfork core-868-890",
-               ("asset_sym", current_asset.symbol)("asset_id", current_asset.id) );
```
Comment (abitmore) [inline libraries/chain/db_maint.cpp]: I guess we've found something, so this function can't be removed? Please update the TODO above to a comment explaining why this function can't be removed.

Comment (jmjatlanta) [reply libraries/chain/db_maint.cpp]: Yes, something was found. But the TODO below indicates more can be cleaned up. I am going to have to take a step back and build a better understanding of HF 935. Thanks for the careful review. More to come.

File: libraries/chain/market_evaluator.cpp (hunk 2 above)
Comment (jmjatlanta) [reply libraries/chain/market_evaluator.cpp]: It looks as if at hardfork time, current_supply was below max_supply on all assets. I am verifying. If so, the one-time processing code can be removed. I am also getting very tired of abitmore always being right.

File: libraries/chain/asset_evaluator.cpp [hunk 11]
```diff
@@ -559,24 +559,12 @@ static bool update_bitasset_object_options(
...
-      // We need to call check_call_orders if the price feed changes after hardfork core-935
-      if( next_maint_time > HARDFORK_CORE_935_TIME )
-         return ( !( old_feed == bdo.current_feed ) );
-
       // We need to call check_call_orders if the settlement price changes after hardfork core-868-890
       if( after_hf_core_868_890 )
       {
-         if( old_feed.settlement_price != bdo.current_feed.settlement_price )
+         if( old_feed.settlement_price != bdo.current_feed.settlement_price ||
+               !( old_feed == bdo.current_feed ) )
```
Comment (abitmore) [inline libraries/chain/asset_evaluator.cpp]: Better directly `return ( !( old_feed == bdo.current_feed ) );`? Because `old_feed.settlement_price != bdo.current_feed.settlement_price` is a subset of `!( old_feed == bdo.current_feed )`.

File: libraries/chain/db_market.cpp [hunk 12]
```diff
@@ -791,8 +759,7 @@ asset database::match( const call_order_object& call,
    if( before_core_hardfork_342 )
    {
       auto call_collateral = call.get_collateral();
-      if( call_pays == call_collateral ) // TODO remove warning after hard fork core-342
-         wlog( "Incorrectly captured black swan event at block #${block}", ("block",head_block_num()) );
+      // NOTE: incorrectly captured a black swan event. Unfortunately, this happened.
```
Comment (abitmore) [inline libraries/chain/db_market.cpp]: No, this didn't happen.

File: libraries/chain/db_market.cpp [hunk 13]
```diff
@@ -76,11 +76,8 @@ void database::globally_settle_asset( const asset_object& mia, const price& sett
    {
       if( before_core_hardfork_342 )
       {
+         //Note: If here, the call order may be paying nothing (Yes, this happened)
```
Comment (abitmore) [inline libraries/chain/db_market.cpp]: This didn't happen.

File: libraries/chain/db_market.cpp [hunk 14]
```diff
@@ -1130,11 +1087,8 @@ bool database::check_call_orders( const asset_object& mia, bool enable_black_swa
 
           if( before_core_hardfork_342 )
           {
+             // NOTE: To be here, limit order could paying something for nothing (Yes, this happened)
```
Comment (abitmore) [inline libraries/chain/db_market.cpp]: This didn't happen.

File: tests/tests/bitasset_tests.cpp [hunk 15]
```diff
@@ -933,16 +933,11 @@ BOOST_AUTO_TEST_CASE( hf_935_test )
       int blocks = 0;
       auto mi = db.get_global_properties().parameters.maintenance_interval;
 
-      if( i == 2 ) // go beyond hard fork 890
+      if( i == 2) // go beyond hard fork 890 (which is the same as 935 BTW)
```
Comment (abitmore) [inline tests/tests/bitasset_tests.cpp]: I guess we can remove this test case directly if it's to test hf935 only. If it's used to test hf1270 as well but not documented, better rename the test case and update the document above.

File: libraries/chain/db_maint.cpp [hunk 16]
```diff
@@ -951,7 +949,6 @@ void database::process_bitassets()
 void process_hf_1465( database& db )
 {
    const auto head_num = db.head_block_num();
-   wlog( "Processing hard fork core-1465 at block ${n}", ("n",head_num) );
```
Comment (abitmore) [inline libraries/chain/db_maint.cpp]: Note: this function changed nothing in mainnet, so perhaps remove it, but perhaps need it for testnet.

File: libraries/chain/db_maint.cpp (hunk 10 above)
Comment (jmjatlanta) [reply libraries/chain/db_maint.cpp]: I have verified that this is not needed, and has been removed. More cleanup in the works.

File: libraries/chain/asset_evaluator.cpp (hunk 11 above)
Comment (jmjatlanta) [reply libraries/chain/asset_evaluator.cpp]: I left that in with the intent to evaluate the == operator of the feed later. And then I promptly forgot. It has been fixed.

File: libraries/chain/db_market.cpp (hunk 13 above)
Comment (jmjatlanta) [reply libraries/chain/db_market.cpp]: Hrmmm. I just ran it again, and found 8 instances on mainnet where "pays" is zero. Am I misunderstanding the code (or perhaps misunderstanding your comment)?

Comment (abitmore) [reply libraries/chain/db_market.cpp]: Here is a list of all log messages related to #184 before the hf (produced by replaying with `3.3.1`):
TL;DR: some variant C, some variant F
```
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #17167173                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #19843236                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #22370884                 db_market.cpp:750
th_a:?unnamed?             do_apply ] Something for nothing issue (#184, variant F) occurred at block #22525298                 asset_evaluator.cpp:879
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #23592820                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #23593831                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #23676128                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #24184740                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #24310110                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #25264776                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #25319453                 db_market.cpp:750
th_a:?unnamed?             do_apply ] Something for nothing issue (#184, variant F) occurred at block #25369774                 asset_evaluator.cpp:879
th_a:?unnamed?             do_apply ] Something for nothing issue (#184, variant F) occurred at block #25727532                 asset_evaluator.cpp:879
th_a:?unnamed?             do_apply ] Something for nothing issue (#184, variant F) occurred at block #25930044                 asset_evaluator.cpp:879
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #26155800                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #26722712                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #26722721                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #26722749                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #27011422                 db_market.cpp:750
th_a:?unnamed?             do_apply ] Something for nothing issue (#184, variant F) occurred at block #27352798                 asset_evaluator.cpp:879
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #27887880                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #27887886                 db_market.cpp:750
th_a:?unnamed?                match ] Something for nothing issue (#184, variant C) occurred at block #27936676                 db_market.cpp:750
```

File: libraries/chain/db_market.cpp (hunk 14 above)
Comment (jmjatlanta) [reply libraries/chain/db_market.cpp]: Agreed. Removed the comment.

File: libraries/chain/db_maint.cpp (hunk 16 above)
Comment (jmjatlanta) [reply libraries/chain/db_maint.cpp]: Yes. Needed for testnet. Added comment.

File: libraries/chain/db_market.cpp (hunk 12 above)
Comment (jmjatlanta) [reply libraries/chain/db_market.cpp]: Correct. Comment removed.

File: tests/tests/bitasset_tests.cpp (hunk 15 above)
Comment (jmjatlanta) [reply tests/tests/bitasset_tests.cpp]: It is used to test 1270. Renamed the test and modified the comment above the method.

File: libraries/chain/db_market.cpp (hunk 13 above)
Comment (jmjatlanta) [reply libraries/chain/db_market.cpp]:
```
//Note: If here, the call order may be paying nothing (Yes, this happened)
pays = call_itr->get_debt() * settlement_price; // round down, in favor of call order
if ( pays.amount == 0 )
{
     wlog("Before HF 342, paying nothing for collateral ${asset} at block ${blk}", 
            ("asset", call_itr->get_collateral())
            ("blk", head_block_num())
       );
}
```
results in:
```
default.log:2019-09-10T21:42:50        th_a:?unnamed? globally_settle_asse ] Before HF 342, paying nothing for collateral {"amount":30000000,"asset_id":"1.3.0"} at block 14032429			db_market.cpp:86
default.log:2019-09-10T21:42:50        th_a:?unnamed? globally_settle_asse ] Before HF 342, paying nothing for collateral {"amount":"2264664694","asset_id":"1.3.0"} at block 14032429			db_market.cpp:86
default.log:2019-09-10T21:42:50        th_a:?unnamed? globally_settle_asse ] Before HF 342, paying nothing for collateral {"amount":149600000,"asset_id":"1.3.0"} at block 14032429			db_market.cpp:86
default.log:2019-09-10T21:42:58        th_a:?unnamed? globally_settle_asse ] Before HF 342, paying nothing for collateral {"amount":2000000000,"asset_id":"1.3.0"} at block 14689927			db_market.cpp:86
default.log:2019-09-10T21:42:58        th_a:?unnamed? globally_settle_asse ] Before HF 342, paying nothing for collateral {"amount":290000000,"asset_id":"1.3.0"} at block 14689927			db_market.cpp:86
default.log:2019-09-10T21:42:58        th_a:?unnamed? globally_settle_asse ] Before HF 342, paying nothing for collateral {"amount":100000000,"asset_id":"1.3.0"} at block 14689927			db_market.cpp:86
default.log:2019-09-10T21:42:58        th_a:?unnamed? globally_settle_asse ] Before HF 342, paying nothing for collateral {"amount":935875108,"asset_id":"1.3.0"} at block 14689927			db_market.cpp:86
default.log:2019-09-10T21:42:58        th_a:?unnamed? globally_settle_asse ] Before HF 342, paying nothing for collateral {"amount":1165000000,"asset_id":"1.3.0"} at block 14689927			db_market.cpp:86
```

**Comment by pmconrad** [comment]
Please rebase on latest develop to resolve the conflict.

File: libraries/chain/db_market.cpp (hunk 13 above)
Comment (abitmore) [reply libraries/chain/db_market.cpp]: Prediction markets which have `0` as outcome can pay nothing for the collateral. That's why there was a check for PM there.

Comment (jmjatlanta) [reply libraries/chain/db_market.cpp]: So, I guess there is no need for a comment here, as this is normal. I will rearrange.

**Comment by pmconrad** [comment]
Something broke during rebase.

File: libraries/chain/db_maint.cpp [hunk 17]
```diff
@@ -1059,88 +1051,12 @@ void process_hf_868_890( database& db, bool skip_check_call_orders )
...
-      if( feeds_changed )
-      {
-         wlog( "Found invalid feed for asset ${asset_sym} (${asset_id}) during hardfork core-868-890",
-               ("asset_sym", current_asset.symbol)("asset_id", current_asset.id) );
-      }
-
       // always update the median feed due to https://github.com/bitshares/bitshares-core/issues/890
       db.modify( bitasset_data, [head_time,next_maint_time]( asset_bitasset_data_object &obj ) {
          obj.update_median_feeds( head_time, next_maint_time );
       });
```
Comment (abitmore) [inline libraries/chain/db_maint.cpp]: Please add a note after `update_median_feeds`: normally we should call `check_call_orders()` after called `update_median_feeds()`, but for mainnet actually `check_call_orders()` would do nothing, so we skipped it for better performance.

**Comment by abitmore** [comment]
Please resolve conflicts (again).

File: tests/common/database_fixture.cpp [hunk 18]
```diff
@@ -94,7 +94,8 @@ database_fixture::database_fixture(const fc::time_point_sec &initial_timestamp)
 
    genesis_state.initial_timestamp = initial_timestamp;
 
-   if(current_test_name == "hf_935_test") {
+   if(boost::unit_test::framework::current_test_case().p_name.value == "hf_1270_test")
```
Comment (abitmore) [inline tests/common/database_fixture.cpp]: Please only change the test case name here.
